### DISCUSSION TITLE
code-server as a browser add-on

### DISCUSSION THREAD

**Discussion by sergei3000** [coder/code-server#2047] [Ideas]
Are there any plans to create browser add-ons/extensions/apps for popular browsers?

Could be a great option for the cases when users don't have sudo rights on their machines where they want to use code-server or where the operating system is not supported by code-server.

**Comment by code-asher** [comment]
We don't have any plans for a browser extension at the moment. Could you clarify what the extension would do? code-server would still need to run a backend which means an extension wouldn't help with operating system support.

It might be possible to run the backend in a browser one day but at the moment I don't think that's possible.

You should be able to run code-server as non-root already, by the way.

**Comment by sr229** [comment]
Coder had [Sail](https://sail.dev/) which is what you're probably looking for - but it has been long deprecated.

**Comment by sergei3000** [comment]
@code-asher yeah, I meant running backend in a browser

**Reply by code-asher** [reply]
Gotcha. It's definitely a neat idea. Could be interesting to hack on it and see how far we get but I suspect the state of extensions right now won't let us do everything we need (for example spawning child processes).

**Reply by code-asher** [reply]
Actually I think we could get pretty far. We could use web workers for separate threads. We already shim the Node API so we can run extensions on the browser so we'd just need to modify that to work directly in the browser instead of calling to a Node backend. This actually might be pretty doable.

**Reply by JammSpread** [reply]
@code-asher If that doesn't work you could make a custom version of chromium with code-server running in the background. Though might not make much sense.

**Reply by sr229** [reply]
> @code-asher If that doesn't work you could make a custom version of chromium with code-server running in the background. Though might not make much sense.

why would you want that? There's already VS Code for that purpose.
